examples/capture/main.rs

```diff
@@ -0,0 +1,99 @@
+/// the added benefit that this method doesn't require a window to be created.
+use wgpu::*;
+
+fn main() {
+    env_logger::init();
+
+    let instance = Instance::new();
+
+    let adapter = instance.get_adapter(&AdapterDescriptor {
+        power_preference: PowerPreference::LowPower,
+    });
+
+    let mut device = adapter.request_device(&DeviceDescriptor {
+        extensions: Extensions {
+            anisotropic_filtering: false,
+        },
+        limits: Limits::default(),
+    });
+
+    // Rendered image is 256×256 with 32-bit RGBA color
+    let size = 256u32;
+
+    // The output buffer lets us retrieve the data as an array
+    let output_buffer = device.create_buffer(&BufferDescriptor {
+        size: (size * size * std::mem::size_of::<u32>() as u32) as u64,
+        usage: BufferUsage::MAP_READ | BufferUsage::TRANSFER_DST,
+    });
+
+    let texture_extent = Extent3d {
+        width: size,
+        height: size,
+        depth: 1,
+    };
+
+    // The render pipeline renders data into this texture
+    let texture = device.create_texture(&TextureDescriptor {
+        size: texture_extent,
+        array_layer_count: 1,
+        mip_level_count: 1,
+        sample_count: 1,
+        dimension: TextureDimension::D2,
+        format: TextureFormat::Rgba8Unorm,
+        usage: TextureUsage::STORAGE | TextureUsage::OUTPUT_ATTACHMENT | TextureUsage::TRANSFER_SRC,
+    });
+
+    // Set the background to be red
+    let command_buffer = {
+        let mut encoder =
+            device.create_command_encoder(&CommandEncoderDescriptor { todo: 0 });
+        encoder.begin_render_pass(&RenderPassDescriptor {
+            color_attachments: &[RenderPassColorAttachmentDescriptor {
+                attachment: &texture.create_default_view(),
+                resolve_target: None,
+                load_op: LoadOp::Clear,
+                store_op: StoreOp::Store,
+                clear_color: Color::RED,
+            }],
+            depth_stencil_attachment: None,
+        });
+
+        // Copy the data from the texture to the buffer
+        encoder.copy_texture_to_buffer(
+            TextureCopyView {
+                texture: &texture,
+                mip_level: 0,
+                array_layer: 0,
+                origin: Origin3d::ZERO,
+            },
+            BufferCopyView {
+                buffer: &output_buffer,
+                offset: 0,
+                row_pitch: std::mem::size_of::<u32>() as u32 * size,
+                image_height: size,
+            },
+            texture_extent,
+        );
+
+        encoder
+    }
+    .finish();
+
+    device.get_queue().submit(&[command_buffer]);
+
+    // Write the buffer as a PNG
+    output_buffer.map_read_async(
+        0,
+        (std::mem::size_of::<u32>() as u32 * size * size) as u64,
+        move |result: BufferMapAsyncResult<&[u8]>| {
+            let png_encoder = image::png::PNGEncoder::new(std::fs::File::create("red.png").unwrap());
```

Comment:
could we only use the `png` crate directly (instead of `image`) in order to reduce the dev build times?

Reply:
Yes, no problem